core/src/transaction_scheduler.rs

```diff
@@ -0,0 +1,1631 @@
+        scheduler_request: SchedulerRequest,
+    ) {
+        let response_sender = scheduler_request.response_sender;
+        match scheduler_request.msg {
+            SchedulerMessage::RequestBatch { num_txs, bank } => {
+                trace!("SchedulerMessage::RequestBatch num_txs: {}", num_txs);
+                let (sanitized_transactions, rescheduled_packets) = Self::get_scheduled_batch(
+                    unprocessed_packets,
+                    scheduled_accounts,
+                    num_txs,
+                    &bank,
+                );
+                trace!(
+                    "sanitized_transactions num: {}, rescheduled_packets num: {}, unprocessed_packets num: {}",
+                    sanitized_transactions.len(),
+                    rescheduled_packets.len(),
+                    unprocessed_packets.len()
+                );
+
+                let _ = response_sender
+                    .send(SchedulerResponse::ScheduledBatch(ScheduledBatch {
+                        sanitized_transactions,
+                    }))
+                    .unwrap();
+
+                // push rescheduled back on
+                for tx in rescheduled_packets {
+                    unprocessed_packets.push(tx.clone());
+                }
+            }
+            SchedulerMessage::Ping { id } => {
+                let _ = response_sender
+                    .send(SchedulerResponse::Pong(Pong { id }))
+                    .unwrap();
+            }
+            SchedulerMessage::ExecutedBatchUpdate {
+                executed_transactions,
+                rescheduled_transactions,
+            } => {
+                {
+                    // drop account locks
+                    let mut account_locks = scheduled_accounts.lock().unwrap();
+                    for tx in executed_transactions
+                        .iter()
+                        .chain(rescheduled_transactions.iter())
+                    {
+                        let tx_locks = tx.get_account_locks_unchecked();
+                        trace!("unlocking locks: {:?}", tx_locks);
+                        Self::drop_account_locks(
+                            &mut account_locks,
+                            &tx_locks.writable,
+                            &tx_locks.readonly,
+                        );
+                    }
+                    trace!("dropped account locks, account_locks: {:?}", account_locks);
+                }
+                // TODO (LB): reschedule transactions as packet
+                // for tx in rescheduled_transactions {
+                //     unprocessed_packets.push(tx.deser)
+                // }
+                let _ = response_sender
+                    .send(SchedulerResponse::ExecutedBatchResponse(
+                        ExecutedBatchResponse {},
+                    ))
+                    .unwrap();
+            }
+        }
+    }
+
+    fn can_lock_accounts(
+        account_locks: &AccountLocks,
```

Comment:
curious why not wrap this in a `MutexGuard` to be explicit that call-site has acquired a lock on this object?